api/controllers/game/spectate/join.js

```diff
@@ -65,8 +65,12 @@ module.exports = async function (req, res) {
     }
     const gameState = unpackGamestate(game.gameStates.at(gameStateIndex));
     const socketEvent = await createSocketEvent(game, gameState);
-    Game.publish([ game.id ], socketEvent);
-
+    // Only notify others that a spectator joined, do not send full game state
+    Game.publish([ game.id ], {
+      gameId: game.id,
+      change: 'spectatorJoined',
+      username: spectator.username,
+    });
```

Comment:
This is the core bug fix. We were publishing an event with the full game data and handling it by updating the game state for all subscribed clients, even when the spectator is looking at an old game state and other subscribers are looking at latest. 

Clients should instead only update the list of spectators by adding the new one to the list